### PR TITLE
Write detailed Diff-Drive-Controller documentation to make all the interfaces understandable.

### PR DESCRIPTION
**Pull request by destogl**
Please help where TODO are added.

This is not yet finished, but provides an initial structure.

### CODE REVIEW

File: diff_drive_controller/doc/userdoc.rst [hunk 1]
```diff
@@ -24,3 +24,135 @@ Other features
...
+
+left_wheel_radius_multiplier [double]
+  Correction factor when radius of left wheels differs from the nominal value in ``wheel_radius`` parameter.
+
+right_wheel_radius_multiplier [double]
+  Correction factor when radius of right wheels differs from the nominal value in ``wheel_radius`` parameter.
+
+Odometry
+,,,,,,,,,
+odom_frame_id [string] (default: "base_link")
```
Comment (destogl) [inline diff_drive_controller/doc/userdoc.rst:106]:
```suggestion
base_frame_id [string] (default: "base_link")
```

File: diff_drive_controller/doc/userdoc.rst [hunk 2]
```diff
@@ -24,3 +24,135 @@ Other features
...
+  Name of the frame for odometry.
+  This frame is parent of ``base_frame_id`` when controller publishes odometry.
+
+base_frame_id [string] (default: "odom")
```
Comment (destogl) [inline diff_drive_controller/doc/userdoc.rst:110]:
```suggestion
odom_frame_id [string] (default: "odom")
```

File: diff_drive_controller/doc/userdoc.rst [hunk 3]
```diff
@@ -24,3 +24,135 @@ Other features
...
+
+Commands
+```````````
+cmd_vel_timeout [double] (default: "0.5s")
+  Timeout after which input command on ``cmd_vel`` topic is considered staled.
+
+publish_limited_velocity [double] (default: "false")
+  Publish limited velocity value.
+
+
```
Comment (bmagyar) [inline diff_drive_controller/doc/userdoc.rst:143]: extra newline?

File: diff_drive_controller/doc/userdoc.rst [hunk 4]
```diff
@@ -24,3 +24,135 @@ Other features
...
+velocity_rolling_window_size [int] (default: "10")
+  (TODO(destogl): Please help me describe this correctly)
+
+Commands
+```````````
+cmd_vel_timeout [double] (default: "0.5s")
+  Timeout after which input command on ``cmd_vel`` topic is considered staled.
+
+publish_limited_velocity [double] (default: "false")
```
Comment (bmagyar) [inline diff_drive_controller/doc/userdoc.rst:140]: these are parameters, no?

File: diff_drive_controller/doc/userdoc.rst [hunk 5]
```diff
@@ -24,3 +24,135 @@ Other features
...
+position_feedback [bool] (default: "true")
+  Is there position feedback from hardware.
+
+enable_odom_tf [bool] (default: "true")
+  Publish transformation between ``odom_frame_id`` and ``base_frame_id``.
+
+velocity_rolling_window_size [int] (default: "10")
+  (TODO(destogl): Please help me describe this correctly)
+
+Commands
```
Comment (bmagyar) [inline diff_drive_controller/doc/userdoc.rst:135]: this is slightly misleading: on first impression I thought these are the topics to command the controller, the rst heading should help it though when rendered

**Review by bmagyar — APPROVED** [review]

File: diff_drive_controller/doc/userdoc.rst [hunk 6]
```diff
@@ -24,3 +24,135 @@ Other features
     Odometry publishing
     Task-space velocity, acceleration and jerk limits
     Automatic stop after command time-out
+
+
+ros2_control Interfaces
```
Comment (bmagyar) [inline diff_drive_controller/doc/userdoc.rst:29]: I think people are mostly going to be interested in the ROS2 interfaces and parameters, the ros2_control interfaces should have a much smaller audience so I'm proposing to shift this section below the others

**Pull request merged**